### PR TITLE
Chore/json reader implementation

### PR DESCRIPTION
**Pull request by sebastian-barra-jalafund**
## Description

extracts information from .json files and saves them in a list of objects

## Type of Change

Put an `x` in all the boxes that apply:

- [ ] ✨ New Feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug Fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking Change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code Refactor
- [ ] ✅ Build Configuration Change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

## Pull Request Checklist

Put an `x` in all the boxes that apply:

- [ ] My commit messages are detailed.
- [ ] My code follows the code style of this project.
- [ ] No existing features have been broken without good reason.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests pass.

### CODE REVIEW

File: src/main/java/org/example/fuzzy_matcher/util/DataReader.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package org.example.fuzzy_matcher.util;
```
Comment (teran-joseluis) [inline src/main/java/org/example/fuzzy_matcher/util/DataReader.java]: @SebasBarra02 This class should be out of the fuzzy matcher library due to it's not part of it, so please create a new directory and move it.

File: src/main/java/org/example/fuzzy_matcher/util/DataReader.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            File jsonFile = new File(filePath);
+            ObjectMapper objectMapper = new ObjectMapper();
+            return objectMapper.readValue(jsonFile, UserList.class);
+        } catch (IOException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
```
Comment (teran-joseluis) [inline src/main/java/org/example/fuzzy_matcher/util/DataReader.java]: @SebasBarra02 This method shouldn't be here. Move it to the main we have or if you want to test your previous methods, you should write unit tests for it.

File: src/main/java/org/example/objects/MusicList.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package org.example.objects;
```
Comment (teran-joseluis) [inline src/main/java/org/example/objects/MusicList.java]: @SebasBarra02 To ensure the functionality of this class and if more methods will create it here, you should create unit tests, so please review it.

File: src/main/java/org/example/objects/User.java [hunk 4]
```diff
@@ -1,30 +1,36 @@
...
   private int age;
-  private List<String> favoriteGenres;
-  private List<String> favoriteArtists;
-  private List<String> favoriteSongs;
-  private List<String> recentlyPlayed;
-
-  public User(int id, String name, int age, List<String> favoriteGenres,
-              List<String> favoriteArtists, List<String> favoriteSongs,
-              List<String> recentlyPlayed) {
+  private List<String> favorite_genres;
```
Comment (teran-joseluis) [inline src/main/java/org/example/objects/User.java]: @SebasBarra02 Can you explain me why the change of the variable convention name? Is it was really necessary?
Let me known that but you should maintain the one we have before.

File: src/main/java/org/example/objects/User.java [hunk 5]
```diff
@@ -57,18 +63,18 @@ public int getAge() {
   }
 
   public List<String> getFavoriteGenres() {
-    return this.favoriteGenres;
```
Comment (teran-joseluis) [inline src/main/java/org/example/objects/User.java]: @SebasBarra02 The same question here and for the rest too.

File: src/main/java/org/example/objects/UserList.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
+package org.example.objects;
```
Comment (teran-joseluis) [inline src/main/java/org/example/objects/UserList.java]: @SebasBarra02 As I mentioned on MusicList.java, please review that.

Comment (sebastian-barra-jalafund) [reply src/main/java/org/example/objects/UserList.java]: test units were added.

File: src/main/java/org/example/objects/MusicList.java (hunk 3 above)
Comment (sebastian-barra-jalafund) [reply src/main/java/org/example/objects/MusicList.java]: test units were added

File: src/main/java/org/example/objects/User.java [hunk 7]
```diff
@@ -1,30 +1,39 @@
...
-
-  public User(int id, String name, int age, List<String> favoriteGenres,
-              List<String> favoriteArtists, List<String> favoriteSongs,
-              List<String> recentlyPlayed) {
+  @JsonProperty("favorite_genres")
+  private List<String> favorite_genres;
+  @JsonProperty("favorite_artists")
+  private List<String> favorite_artists;
+  private List<String> favorite_songs;
+  private List<String> recently_played;
```
Comment (Misax148) [inline src/main/java/org/example/objects/User.java]: We need to keep the camel case convention, please don't change the attribute syntax unless it is really necessary

File: src/main/java/org/example/objects/User.java [hunk 8]
```diff
@@ -1,30 +1,39 @@
...
+import com.fasterxml.jackson.annotation.JsonProperty;
 import org.example.fuzzy_matcher.domain.Document;
 import org.example.fuzzy_matcher.domain.Element;
 import org.example.fuzzy_matcher.domain.ElementType;
 
 import java.util.List;
 
+@JsonIgnoreProperties(ignoreUnknown = true)
 public class User {
   private int id;
```
Comment (Misax148) [inline src/main/java/org/example/objects/User.java:13]: I have a question, is the id user in this attribute loaded from the json? The id is important for Match Service

File: src/main/java/org/example/objects/User.java [hunk 9]
```diff
@@ -1,30 +1,39 @@
...
   private String name;
```
Comment (Misax148) [inline src/main/java/org/example/objects/User.java:14]: The same question with that attribute

File: src/main/java/org/example/objects/User.java [hunk 10]
```diff
@@ -1,30 +1,39 @@
...
   private int age;
```
Comment (Misax148) [inline src/main/java/org/example/objects/User.java:15]: The same question with that attribute

File: src/main/java/org/example/objects/Music.java [hunk 11]
```diff
@@ -1,22 +1,43 @@
...
 import org.example.fuzzy_matcher.domain.Document;
 import org.example.fuzzy_matcher.domain.Element;
 import org.example.fuzzy_matcher.domain.ElementType;
 
+
 import java.util.List;
 
+@JsonIgnoreProperties(ignoreUnknown = true)
 public class Music {
   private int id;
```
Comment (Misax148) [inline src/main/java/org/example/objects/Music.java]: I have a question, is the id music in this attribute loaded from the json? The id is important for Match Service

File: src/main/java/org/example/objects/User.java (hunk 8 above)
Comment (sebastian-barra-jalafund) [reply src/main/java/org/example/objects/User.java:13]: if the input JSON contains additional properties that are not defined in the User class, Jackson will simply ignore them

File: src/main/java/org/example/objects/User.java (hunk 7 above)
Comment (sebastian-barra-jalafund) [reply src/main/java/org/example/objects/User.java]: ok, the variables will not be changed

**Review by Misax148 — APPROVED** [review]

**Review by Misax148 — APPROVED** [review]

**Review by teran-joseluis — APPROVED** [review]
